### ISSUE TITLE
Snapshot is ill documented

### ISSUE THREAD

**Issue by jondequinor** [equinor/configsuite#98]
The snapshot needs its own section under `Getting started`. Currently, the best documentation for the snapshot is under `The main interface`, and it is also partially documented in other sections.

The reason for it needing it own section, is that:

- it is an essential feature in configsuite
- _snapshot_ might be strongly associated with other _snapshotting_ mechanisms from other systems (virtualizations, disk management, etc). Here, snapshotting is an explicit user action, whereas in configsuite is seems like an implicit action as a result of successful validation?

**Comment by markusdregi** [comment]
I completely agree. I even think it should be renamed, as `snapshot` came from the idea that at least that should be immutable. But luckily the implementation made the `ConfigSuite` instance itself immutable.
